libnetwork/drivers/bridge/bridge_linux.go

```diff
@@ -337,6 +340,8 @@ func (c *networkConfiguration) fromLabels(labels map[string]string) error {
 			if c.EnableICC, err = strconv.ParseBool(value); err != nil {
 				return parseErr(label, value, err.Error())
 			}
+		case TrustedHostInterfaces:
+			c.TrustedHostInterfaces = strings.FieldsFunc(value, func(r rune) bool { return r == ':' })
```

Comment:
> FieldsFunc splits the string s at each run **of Unicode code points**...

(emphasis mine.)

`value` is a bag of bytes, not a Unicode string. While I can't come up with an pathological input which would split in an unexpected way, I still worry about the risk of surprising behaviours from treating the value as a UTF-8 string.
```suggestion
			c.TrustedHostInterfaces = strings.Split(value, ":")
```

Reply:
Not quite that, because `strings.Split("", ":")` is [""] ... so the bridge driver sees an empty interface name and generates a broken iptables rule. But, ok.

Reply:
Fuzzing found nothing. My fears were unfounded. `strings.FieldsFunc` is fit for purpose, as far as I can tell, to split a bag-of-bytes string on a UTF-8 codepoint. And it makes sense once I think about it: UTF-8 is self-synchronizing so there is no confusing an ASCII char for part of a multibyte sequence.

Reply:
Ok, thanks Cory - I've put it back to `FieldsFunc`, and rebased just because it had got a bit behind. I'll let the tests run, then merge.